src/app/tracing/util/map-utils.ts

```diff
@@ -119,49 +128,151 @@ export function createShapeFileLayer(
     return vectorLayer;
 }
 
-function createVectorLayerStyle(styleConfig: ShapeMapStyleConfig): StyleLike {
+function createVectorLayerStyle(styleConfig: ShapeStyleSettings): StyleLike {
     return new Style({
         stroke: new Stroke({
             color: [
-                styleConfig.lineColor.r,
-                styleConfig.lineColor.g,
-                styleConfig.lineColor.b,
+                styleConfig.geojsonBorderColor.r,
+                styleConfig.geojsonBorderColor.g,
+                styleConfig.geojsonBorderColor.b,
             ],
-            width: styleConfig.lineWidth,
+            width: styleConfig.geojsonBorderWidth,
         }),
     });
 }
 
-function getMapLayer(map: ol.Map): BaseLayer | null {
-    const layers = map
+function getMapLayer(map: ol.Map, layerType): Array<BaseLayer> {
+    return map
         .getLayers()
         .getArray()
-        .filter((layer) => layer.get(LAYER_ID_KEY) === MAP_LAYER_ID);
-    return layers.length > 0 ? layers[0] : null;
+        .filter((layer) => layer instanceof layerType);
 }
 
 export function updateVectorLayerStyle(
     map: ol.Map,
-    styleConfig: ShapeMapStyleConfig,
+    styleConfig: ShapeFileSettings,
 ): void {
-    const vectorLayer = getMapLayer(map);
-    if (vectorLayer instanceof VectorLayer) {
-        const style = createVectorLayerStyle(styleConfig);
-        vectorLayer.setStyle(style);
-    }
+    const mapLayers = getMapLayer(map, VectorLayer);
+    // loop through all layers
+    mapLayers.forEach((layer) => {
+        // apply styles only to instances of VectorLayer
+        if (layer instanceof VectorLayer) {
+            const style = createVectorLayerStyle(styleConfig);
+            layer.setStyle(style);
+        }
+    });
 }
 
+export const setLayerVisibility = (
+    layer: BaseLayer,
+    visibility: boolean,
+    layerDataHasChanged: boolean,
+    callback,
+): void => {
+    const layerExisits = layer?.setVisible !== undefined;
+    // if layer does not exist and should not be visible, do nothing
+    if (!layerExisits && !visibility) {
+        return;
+    }
+
+    // if layer data has changed or layer does not exist but should be visible, let callback handle creation of layer
+    if ((!layerExisits || layerDataHasChanged) && visibility) {
+        callback();
```

Comment:
Can this callback be named something like `createLayerCallback`?

Reply:
- [ ]  better naming for callback

Reply:
done